### PR TITLE
doc: update docs for new features

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Docs:
- added `plugin-configuration.md` to explain arguments and permission management
- update ca-signed certificate getting started doc

Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

**Comment by JeyJeyGao** [comment]
@Two-Hearts Your [comment](https://github.com/Azure/notation-azure-kv/pull/119#pullrequestreview-1534075066) will be resolved in this PR.

File: docs/ca-signed-workflow.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Sign and verify an artifact with a certificate signed by a trusted CA in Azure Key Vault
 > **Note** The following guide can be executed on Linux bash, macOS Zsh and Windows WSL
 1. [Install the Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli)
-2. Log in to Azure with Azure CLI, set the subscription and make sure the `GetCertificate` and `Sign` permission have been granted to your role:
+2. Log in to Azure with Azure CLI, set the subscription and make sure the `Secrets Get` and `Key Sign` permission have been granted to your role:
```
Comment (FeynmanZhou) [inline docs/ca-signed-workflow.md:4]: I think this permission prerequisite is required for Azure Key Vault, right? Let's be more specific here. Is there a related Azure doc that users can follow to configure or check the permission? If so, we need to add the link here.

File: docs/plugin-config.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+> They must be concatenated such that each certificate directly validates the preceding one. The following example features two certificates: an intermediate and a root certificate, but your certificate chain may include more or fewer.
+>
+> ```pem
+> -----BEGIN CERTIFICATE-----
+> Base64–encoded intermediate certificate
+> -----END CERTIFICATE-----
+> -----BEGIN CERTIFICATE-----
+> Base64–encoded root certificate
+> -----END CERTIFICATE-----
+> ```
```
Comment (Two-Hearts) [inline docs/plugin-config.md:43]: An end user doesn't always have to manually create a cert bundle on their own. Sometimes, they can download a cert bundle online, for example, https://www.microsoft.com/en-us/download/details.aspx?id=102267. We should mention this kind of cert bundle source in the doc.

File: docs/plugin-config.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Plugin Configuration
+The `notation-azure-kv` plugin supports plugin configuration to use advanced features.
```
Comment (Two-Hearts) [inline docs/plugin-config.md:2]: For `notation-azure-kv` plugin, all the plugin configs are for one purpose, i.e., `get a valid cert chain or self-signed cert during remote signing`. Let's make it clear in the doc.

File: docs/ca-signed-workflow.md (hunk 1 above)
Comment (JeyJeyGao) [reply docs/ca-signed-workflow.md]: Updated to include the script to set the permission and the link for AKV access policy.

File: docs/plugin-config.md (hunk 3 above)
Comment (JeyJeyGao) [reply docs/plugin-config.md]: Thanks! Added.

File: docs/plugin-config.md (hunk 2 above)
Comment (JeyJeyGao) [reply docs/plugin-config.md:43]: Updated. The link is Microsoft 365 certificate. I'm not sure whether it is for Microsoft owned product or not. So I change to use the example of digicert.

Comment (Two-Hearts) [reply docs/plugin-config.md:43]: > Updated. The link is Microsoft 365 certificate. I'm not sure whether it is for Microsoft owned product or not. So I change to use the example of digicert.

Yep, I'm just providing a dummy example for your reference.

**Review by Two-Hearts — APPROVED** [review]
LGTM

File: docs/plugin-config.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  --id <key_identifier> \
+  --plugin-config self_signed=true
+```
+
+## Permission management
+The `notation-azure-kv` plugin support multiple level of permissions setting to satisfy different permission use cases.
+
+`Key Sign` permission is requred for remote signing, while `Secrets Get` and `Certificates Get` are optional based on use cases.
+
+| `ca_certs` | `self_signed` | Key Vault Permission | Explain                                                                                            |
```
Comment (yizha1) [inline docs/plugin-config.md:74]: The table is a bit confusing. for example, do users need to set self-signed=false if certificate includes certificate chain?

Maybe we can just list 3 scenarios:
- for certificate with chain, no config is required, permission is `secret get` and `sign`
- for certificate without chain, use config `ca_certs`, permission is `cert get` and `sign`
- for self-signed certificate, using config `self_signed=true`, and permissions required is `cert get` and `sign`.

**Review by yizha1 — COMMENTED** [review]
Thanks @JeyJeyGao

File: docs/plugin-config.md (hunk 4 above)
Comment (JeyJeyGao) [reply docs/plugin-config.md]: Thanks! Updated

File: docs/ca-signed-workflow.md [hunk 5]
```diff
@@ -1,27 +1,40 @@
...
    ```
+   Assign `Secrets Get` and `Key Sign` permission to your credentials:
+   ```sh
+   userId=$(az ad signed-in-user show --query id -o tsv)
+   az keyvault set-policy -n "$keyVault" \
+      --key-permissions sign \
+      --secret-permissions get \
+      --upn "$userId"
+   ```
+   > **Note** The script assigns the permission to the current user, and you can also assign the permission to your [managed identity](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview) or [service principal](https://learn.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals?tabs=browser).
```
Comment (shizhMSFT) [inline docs/ca-signed-workflow.md:32]: nit: GitHub changed the rendering rule
```suggestion
   > **Note**
   > The script assigns the permission to the current user, and you can also assign the permission to your [managed identity](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview) or [service principal](https://learn.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals?tabs=browser).
```

Comment (shizhMSFT) [reply docs/ca-signed-workflow.md:32]: > **Note** Hello world

v.s.

> **Note**
> Hello world

File: docs/plugin-config.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+# Plugin Configuration
+The `notation-azure-kv` plugin supports plugin configuration to get a valid cert chain or self-signed cert during remote signing.
+
+## Usage
+- Add the plugin configuration when adding the key
+  ```sh
+  notation key add \
+    --plugin azure-kv \
+    --id <key_identifier> \
+    --plugin-config <argument>=<value> \
```
Comment (shizhMSFT) [inline docs/plugin-config.md:10]: It renders better if we have quotes
```suggestion
    --plugin-config "<argument>=<value>" \
```

Comment (JeyJeyGao) [reply docs/plugin-config.md]: Updated.

File: docs/ca-signed-workflow.md (hunk 5 above)
Comment (JeyJeyGao) [reply docs/ca-signed-workflow.md]: Updated to use the new style. 

However, the symbol still cannot be shown in a markdown file. I guess the feature was disabled for markdown files, but it still works for comments.

Comment (JeyJeyGao) [reply docs/ca-signed-workflow.md]: Example:
# Sign and verify an artifact with a certificate signed by a trusted CA in Azure Key Vault
> **Note**
> The following guide can be executed on Linux bash, macOS Zsh and Windows WSL
1. [Install the Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli)

vs
image

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**